### DISCUSSION TITLE
Regarding Course selection

### DISCUSSION THREAD

**Discussion by shashankn75** [snigi-gupta/AttendingUniversityAtBuffalo#4] [Q&A] [answered]
Hi Snigdha,
Had attended your webinar on 10th Jan, it was really helpful.
I have query regarding selection of courses this sem.
I am registering for below classes this spring21-
CSE565 Computer security 
CSE505 Programming system fundamental
CSE586 Distributed systems
CSE707 3cr Seminar- Security in Emerging Computing

Are they a mix of easy and difficult courses?
I wish to take up computer security domain, if you have any suggestions for me.

**Comment by snigi-gupta** [comment, marked as answer]
@shashankn75,
CSE 565 or Marina's CS is defnitely medium course. She is a strict teacher so definitely make sure you do not plagiarize even a sentence from the internet. She does not grant any mistakes. Also, check out the course review [here](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/blob/main/Computer%20Science%20Important%20Tips/CourseReview.md)

CSE 505 is something which is new and offered for your batch. Although Jayraman is a really good professor. His another course OOAD is a very popular elective and his teaching is greatly appreciated. I think this course is Easy level.

CSE 586 by Blanton should be good. Up until last Spring semester Steve ko used to take it. His course was very tough and I am assuming so should be Blanton's. Good choice since you will work in Go Lang. There is demand for Go in the market. I would say go for it. Hard difficulty subject but definitely worth it.

CSE 707 ..make sure you have the requirements cleared for getting S(satisfactory) for the seminar. Is there a project element to the seminar? or is most reading and summarizing papers? Audit the class.

All in all I think your course structure for SDE specialization looks good with concentration in CompSec. Go for it. You got this!

**Reply by shashankn75** [reply]
Thank you for the clarification.

Yes, CSE707 would have a project component as I would be registering it for 3 credits.

Apart from Leetcode any other suggestions, what kind of skills required for CompSec domain.
I've worked on support projects while at Infosys and have more Linux experience than professional programming. Need your review on that.

**Reply by snigi-gupta** [reply]
@shashankn75 
I think read CS blogs and keep upto date with new technologies. Would definitely suggest pursuing any side projects showcasing them on github. you can find tons of support for that on the internet. It can be a project that someone has already done it. but doing it yourself brings lot of clarity.
having hands on experience in working in multiple OS is definitely a plus. So be sure to market that well. 
Keep your linked in updated and if possible create a portfolio website.

**Reply by shashankn75** [reply]
Okay thank you,

A guide in direction is always helpful.
Thank you again for your efforts and keep up the good work 🙂.

Hoping for the best. Will bug you again wherever stuck.
